### PR TITLE
Implement scoring section 'Package conventions'

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: test/goldens/end2end/pub_server-0.1.4-2.json [hunk 1]
```diff
@@ -904,6 +904,12 @@
         "grantedPoints": 20,
         "maxPoints": 20,
         "summary": "Package gets 20 points if its Dart sdk constraint allows Dart 2."
+      },
+      {
+        "title": "Packaging conventions",
+        "grantedPoints": 15,
+        "maxPoints": 20,
+        "summary": "Package gets 10 points for a valid pubspec.yaml and 5 points for each of README.md and CHANGELOG\n\nNo CHANGELOG found.\nChangelog entries help developers follow the progress of your package. See the [example](https://raw.githubusercontent.com/dart-lang/stagehand/master/templates/package-simple/CHANGELOG.md) generated by `stagehand`."
```
Comment (kwalrath) [inline test/goldens/end2end/pub_server-0.1.4-2.json:912]: Add a `.` before the `\n\n`? (This affects tests.)

Comment (sigurdm) [reply test/goldens/end2end/pub_server-0.1.4-2.json]: Done!

File: lib/src/markdown_content.dart [hunk 2]
```diff
@@ -9,67 +9,56 @@ import 'package:markdown/markdown.dart';
...
+          .querySelectorAll('img')
+          .where((e) => e.attributes.containsKey('src'))
+          .map((e) => Link(e.attributes['src'], e.sourceSpan))),
+      links: _unique(html
+          .querySelectorAll('a')
+          .where((e) => e.attributes.containsKey('href'))
+          .map((e) => Link(e.attributes['href'], e.sourceSpan))));
 }
 
+List<T> _unique<T>(Iterable<T> l) => [
```
Comment (isoos) [inline lib/src/markdown_content.dart:45]: nit: `=> l.toSet().toList()` would be easier to parse for me

File: test/goldens/end2end/angular_components-0.10.0.json [hunk 3]
```diff
@@ -45991,6 +45991,12 @@
         "grantedPoints": 20,
         "maxPoints": 20,
         "summary": "Package gets 20 points if its Dart sdk constraint allows Dart 2."
+      },
+      {
+        "title": "Packaging conventions",
+        "grantedPoints": 15,
+        "maxPoints": 20,
+        "summary": "Package gets 10 points for a valid pubspec.yaml and 5 points for each of README.md and CHANGELOG.md.\n\nLinks in README.md should be secure. 1 link is insecure.\n`README.md:51:22`\n\n```\n   ╷\n51 │ component's provided <a href=\"http://sass-lang.com/guide#topic-6\">Sass mixins</a>.</p>\n   │                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   ╵\n```\n\nUse `https` URLs instead."
```
Comment (isoos) [inline test/goldens/end2end/angular_components-0.10.0.json:45999]: I'm wondering if it is important to provide the full context of the link, maybe the link itself would be enough?

File: lib/src/markdown_content.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/markdown_content.dart:45]: Done

File: test/goldens/end2end/angular_components-0.10.0.json (hunk 3 above)
Comment (sigurdm) [reply test/goldens/end2end/angular_components-0.10.0.json:45999]: But isn't it nice to get it with context?

Comment (isoos) [reply test/goldens/end2end/angular_components-0.10.0.json:45999]: Yeah, let's keep it and fix only when there is a problem with that.

**Pull request merged**